doc/dev/libcephfs_proxy.rst

```diff
@@ -0,0 +1,287 @@
+Design of the libcephfs proxy
+=============================
+
+Description of the problem
+--------------------------
+
+When an application connects to a Ceph volume through the *libcephfs.so*
+library, a cache is created locally inside the process. The *libcephfs.so*
+implementation already deals with memory usage of the cache and adjusts it so
+that it doesn't consume all the available memory. However, if multiple
+processes connect to CephFS through different instances of the library, each
+one of them will keep a private cache. In this case memory management is not
+effective and it's relatively easy to consume all memory when all processes are
+using data cache intensively. This causes the OOM killer to terminate those
```

Comment:
The reconfigurability of libcephfs.so is already stated in the previous sentence. Here it talks about using multiple independent instances of libcephfs, and in this configuration it's pretty easy to trigger the OOM killer. Even configuring the memory limits of each instance it's not possible to avoid consuming all the memory since the number of instances is unbounded.

Anyway, I'll rewrite it to try to explain it in more detail.